webapp/channels/src/components/announcement_bar/version_bar/version_bar.tsx

```diff
@@ -12,67 +12,52 @@ interface Props {
     buildHash?: string;
 }
 
-interface State {
-    buildHashOnAppLoad?: string;
-}
-
-export default class VersionBar extends React.PureComponent <Props, State> {
-    constructor(props: Props) {
-        super(props);
+const VersionBar = ({
+    buildHash,
+}: Props) => {
+    const [buildHashOnAppLoad, setBuildHashOnAppLoad] = useState(buildHash);
 
-        this.state = {
-            buildHashOnAppLoad: props.buildHash,
-        };
-    }
+    const reloadPage = () => {
+        window.location.reload();
+    };
 
-    static getDerivedStateFromProps(props: Props, state: State) {
-        if (!state.buildHashOnAppLoad && props.buildHash) {
-            return {
-                buildHashOnAppLoad: props.buildHash,
-            };
+    useEffect(() => {
+        if (!buildHashOnAppLoad && buildHash) {
+            setBuildHashOnAppLoad(buildHash);
         }
+    }, [buildHash, buildHashOnAppLoad]);
```

Comment:
0/5 I feel this effect only needs to run if the `buildHash` changes. Changes on the `buildHasOnAppLoad` are irrelevant.

I know the linter throws a warning, but it should be fine.